crates/primitives/src/l1/block.rs

```diff
@@ -115,20 +240,41 @@ impl fmt::Debug for L1BlockCommitment {
         write!(
             f,
             "L1BlockCommitment(height={}, blkid={:?})",
-            self.height, self.blkid
+            self.height.to_consensus_u32(),
+            self.blkid
         )
     }
 }
 
 impl L1BlockCommitment {
-    pub fn new(height: u64, blkid: L1BlockId) -> Self {
+    /// Create a new L1 block commitment.
+    ///
+    /// # Arguments
+    /// * `height` - The block height
+    /// * `blkid` - The block ID
+    pub fn new(height: absolute::Height, blkid: L1BlockId) -> Self {
         Self { height, blkid }
     }
 
-    pub fn height(&self) -> u64 {
+    /// Create a new L1 block commitment from a u64 height.
+    ///
+    /// Returns `None` if the height is invalid (greater than u32::MAX).
+    pub fn from_height_u64(height: u64, blkid: L1BlockId) -> Option<Self> {
+        let height = absolute::Height::from_consensus(height as u32).ok()?;
```

Comment:
Honestly it might make sense to "just panic" in this case.  If all calls to this function are all just panics (and there's no sensible error handling path) then there's no point.

Maybe I'm wrong, not sure what the right path to take here is.  I was wrong for trying to be future-proof with using u64 here.